test/functional/ui/float_spec.lua

```diff
@@ -4943,6 +4943,102 @@ describe('floatwin', function()
         ]], float_pos=expected_pos}
         end
       end)
+
+      it("c (:close) closes floats anchored to a window", function()
+        meths.command('tabnew')
+        meths.open_win(0, false, {
+          relative = 'win';
+          width = 1;
+          height = 1;
+          anchor = 'NE';
+          focusable = 0;
+          row = 1;
+          col = 1;
+        })
+        if multigrid then
+          screen:expect{grid=[[
+            ## grid 1
+              {9: }{10:2}{9:+ [No Name] }{3: }{11:2}{3: [No Name] }{5:            }{9:X}|
+              [5:----------------------------------------]|
+              [5:----------------------------------------]|
+              [5:----------------------------------------]|
+              [5:----------------------------------------]|
+              [5:----------------------------------------]|
+              [3:----------------------------------------]|
+            ## grid 2 (hidden)
+              x                                       |
+              {0:~                                       }|
+              {0:~                                       }|
+              {0:~                                       }|
+              {0:~                                       }|
+              {0:~                                       }|
+            ## grid 3
+                                                      |
+            ## grid 4 (hidden)
+              {1:y                   }|
+              {2:~                   }|
+            ## grid 5
+              ^                                        |
+              {0:~                                       }|
+              {0:~                                       }|
+              {0:~                                       }|
+              {0:~                                       }|
+            ## grid 6
+              {1: }|
+            ]], float_pos={
+              [6] = { {
```

Comment:
you can collapse this to one single line for `[6]`

Reply:
Done!